### DISCUSSION TITLE
Module: Access from /server

### DISCUSSION THREAD

**Discussion by go4cas** [nuxt/nuxt#16374] [Questions]
I have a module that starts a socket.io server inside the `listen` hook in `setup`:

```javascript
nuxt.hook('listen', (server) => {
  const io = new Server(server, options.serverOptions)
  options.socketFunctions(io)
})
```

The module also adds a Nuxt plugin, which is accessible by the front-end at runtime, using `$io`.

My question is, how do I access the `io` instance from the server side (either in the `/api`, `/plugins` or `/utilities`)?

PS: I have checked other discussions on socket implementation in Nuxt, but cannot find a solution to access the instance from the server.

**Comment by wobsoriano** [comment]
Your current options are:

1. Attach to `globalThis`
2. Attach to `event.node.req`
